charts/vouch/templates/_helpers.tpl

```diff
@@ -33,12 +33,13 @@ Create chart name and version as used by the chart label.
 
 {{/*
 Common labels
+App version - handle Docker image SHAs by replacing invalid characters and truncate to Kubernetes 63 char limit.
 */}}
 {{- define "vouch.labels" -}}
 helm.sh/chart: {{ include "vouch.chart" . }}
 {{ include "vouch.selectorLabels" . }}
-{{- if .Chart.AppVersion }}
-app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
+{{- if or .Chart.AppVersion .Values.image.tag }}
+app.kubernetes.io/version: {{ .Values.image.tag | default .Chart.AppVersion | replace "@" "." | replace "sha256:" "" | trunc 63 | quote }}
```

Comment:
Agree, this bit is non-standard though @halkeye raised a good point about SHA-based docker images breaking helm charts.

This approach addresses the following error that would result without this change:
```
Error: UPGRADE FAILED: cannot patch "my-vouch" with kind ServiceAccount: ServiceAccount "my-vouch" is invalid: [metadata.labels: Invalid value: "sha256:d34e220de3cfb8a0615ce4554c51b4e03cc22277f28e8984c3047ae7e6a1b949": must be no more than 63 characters, metadata.labels: Invalid value: "sha256:d34e220de3cfb8a0615ce4554c51b4e03cc22277f28e8984c3047ae7e6a1b949": a valid label must be an empty string or consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyValue',  or 'my_value',  or '12345', regex used for validation is '(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?')] && cannot patch "my-vouch" with kind Service: Service "my-vouch" is invalid: [metadata.labels: Invalid value: "sha256:d34e220de3cfb8a0615ce4554c51b4e03cc22277f28e8984c3047ae7e6a1b949": must be no more than 63 characters, metadata.labels: Invalid value: "sha256:d34e220de3cfb8a0615ce4554c51b4e03cc22277f28e8984c3047ae7e6a1b949": a valid label must be an empty string or consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyValue',  or 'my_value',  or '12345', regex used for validation is '(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?')] && cannot patch "my-vouch" with kind Deployment: Deployment.apps "my-vouch" is invalid: [metadata.labels: Invalid value: "sha256:d34e220de3cfb8a0615ce4554c51b4e03cc22277f28e8984c3047ae7e6a1b949": must be no more than 63 characters, metadata.labels: Invalid value: "sha256:d34e220de3cfb8a0615ce4554c51b4e03cc22277f28e8984c3047ae7e6a1b949": a valid label must be an empty string or consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyValue',  or 'my_value',  or '12345', regex used for validation is '(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?')]
```

The main difference is `app.kubernetes.io/version` is consistent with the image tag used. Comparison between 3.2.0 and 3.2.1 tags:
```diff
- ❯ helm upgrade --install my-vouch charts/vouch --set image.tag=latest@sha256:d34e220de3cfb8a0615ce4554c51b4e03cc22277f28e8984c3047ae7e6a1b949
+ ❯ helm upgrade --install my-vouch charts/vouch --set image.tag=latest@sha256:d34e220de3cfb8a0615ce4554c51b4e03cc22277f28e8984c3047ae7e6a1b949
❯ kubectl describe svc my-vouch | grep -C1 version
                   app.kubernetes.io/name=vouch
-                  app.kubernetes.io/version=0.39
+                  app.kubernetes.io/version=latest.d34e220de3cfb8a0615ce4554c51b4e03cc22277f28e8984c3047ae7
-                  helm.sh/chart=vouch-3.2.0
+                  helm.sh/chart=vouch-3.2.1
```

Existing functionality already allows overriding `image.tag` though the there's some inconsistency with the `app.kubernetes.io/version` (and the helm chart `appVersion` - which this PR doesn't address).
Happy to close since functionality already exists. Users should be wary that deployed version may not match `appVersion` and/or `app.kubernetes.io/version` if overridden.